examples/app/bokeh-decision-tree/Decision_Tree/ID3_Decision_Tree/generate_bokeh_data.py

```diff
@@ -0,0 +1,141 @@
+from Decision_Tree.ID3_Decision_Tree.id3_decision_tree import generate_tree
+from Decision_Tree.ID3_Decision_Tree.bucheim import tree_layout
+from Decision_Tree.Plot.instance import Instance
+
+
+def get_depth(node, id_index, visited={}):
+    """ Calculate depth of the tree """
+    if node.decision in Instance().attr_values_dict[Instance().attr_list[-1]]:
+        node.name = Instance().attr_list[-1]
+
+    if not node.children:
+        node.depth = 1
+        # node.id = str(id_index)
+        visited[node] = False
+        id_index += 1
+        return 1
+
+    max_depth = max([get_depth(child, id_index, visited) for child in node.children])
+
+    # node.id = str(id_index)
+    node.depth = max_depth + 1
+    visited[node] = False
+    id_index += 1
+    return max_depth + 1
+
+
+def get_width(node, level):
+    """ Calculate width of the level"""
+    if node is None:
+        return 0
+    if level == 1:
+        return 1
+    elif level > 1:
+        if node.parentPointer:
+            node.depth = node.parentPointer.depth - 1
+        return sum([get_width(child, level-1) for child in node.children])

```

Comment:
You could use `sum(get_width(child, level-1) for child in node.children)` removing the square brackets.